### PR TITLE
Add wcosmo dependency and cosmology module

### PR DESCRIPTION
**Pull request by Qazalbash**

### CODE REVIEW

File: src/pdet/O3.py [hunk 1]
```diff
@@ -85,10 +88,8 @@ def __init__(
         activation = lambda x: jax.nn.leaky_relu(x, 1e-3)
         final_activation = lambda x: (1.0 - 0.0589) * jax.nn.sigmoid(x)
 
-        self.interp_DL = np.logspace(-4, jnp.log10(15.0), 500)
-        self.interp_z = z_at_value(
-            Planck15.luminosity_distance, self.interp_DL * u.Gpc
```
Comment (Qazalbash) [inline src/pdet/O3.py:90]: @zeeshan5885 unit conversion happening here.

File: src/pdet/_emulator.py [hunk 2]
```diff
@@ -212,14 +215,10 @@ def _check_distance(
...
-                list(parameter_dict[COMOVING_DISTANCE]) * u.Gpc,
-            ).value
-            luminosity_distance = (
-                Planck15.luminosity_distance(list(parameter_dict[REDSHIFT]))
-                .to(u.Gpc)
-                .value
+                Planck15.comoving_distance, parameter_dict[COMOVING_DISTANCE]
             )
+            luminosity_distance = Planck15.luminosity_distance(redshift)
+
```
Comment (Qazalbash) [inline src/pdet/_emulator.py:221]: @zeeshan5885 unit conversion happening here.

File: src/pdet/_emulator.py [hunk 3]
```diff
@@ -228,18 +227,13 @@ def _check_distance(
...
+            )
             redshift = jnp.broadcast_to(redshift, shape)
             missing_params[REDSHIFT] = redshift
 
         elif REDSHIFT in parameter_dict:
-            luminosity_distance = (
-                Planck15.luminosity_distance(list(parameter_dict[REDSHIFT]))
-                .to(u.Gpc)
-                .value
-            )
```
Comment (Qazalbash) [inline src/pdet/_emulator.py:242]: @zeeshan5885 unit conversion happening here.

**Comment by Qazalbash** [comment]
with units, we are getting this error, which essentially means the `astropy.units` which is a subclass of `numpy.ndarray` and `jax.numpy.array` are incompatible under jit/grad/vmap transforms.

```bash
Traceback (most recent call last):
  File "/media/gradf/academia/pdet/playground.py", line 35, in <module>
    print(vt())
          ^^^^
  File "/media/gradf/academia/pdet/playground.py", line 22, in vt
    vt_val = VT.predict(
             ^^^^^^^^^^^
  File "/media/gradf/academia/pdet/src/pdet/O3.py", line 176, in predict
    parameter_dict = self.check_input(key, shape, parameter_dict)
                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/media/gradf/academia/pdet/src/pdet/_emulator.py", line 423, in check_input
    key, missing_distance = self._check_distance(key, shape, parameter_dict)
                            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/media/gradf/academia/pdet/src/pdet/_emulator.py", line 221, in _check_distance
    luminosity_distance = Planck15.luminosity_distance(redshift) * units.Gpc
                          ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~
  File "/home/gradf/miniforge3/envs/pdetenv/lib/python3.12/site-packages/quax/_core.py", line 324, in __call__
    out = fn(*args, **kwargs)
          ^^^^^^^^^^^^^^^^^^^
TypeError: unsupported operand type(s) for *: '_QuaxTracer' and 'PrefixUnit'
--------------------
For simplicity, JAX has removed its internal frames from the traceback of the following exception. Set JAX_TRACEBACK_FILTERING=off to include these.
```

**Comment by zeeshan5885** [comment]
@oshaughn Any suggestions to work around this issue?

**Comment by oshaughn** [comment]
Nothing concrete.  You can apparently extend JAX to work with astropy.units, for example https://jax.exoplanet.codes/en/latest/autoapi/jaxoplanet/units/decorator/ .  You can also have something that explicitly switches from astropy units to pint units, as pint units can be compatible with Jax https://pypi.org/project/jpu/

**Pull request merged**